platform/ios/config.cmake

```diff
@@ -1,5 +1,10 @@
 add_definitions(-DMBGL_USE_GLES2=1)
 
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -flto=thin")
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -flto=thin")
+set(CMAKE_EXE_LINKER_FLAGS_RELEASE "${CMAKE_EXE_LINKER_FLAGS_RELEASE} -flto=thin")
+set(CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO} -flto=thin")
```

Comment:
We should consult @kkaefer, but I think using our `set_xcode_property` macro to set Xcode’s `LLVM_LTO` flag will be easier to read in the cmake config and clearer when looking at the settings in Xcode itself.

```
set_xcode_property(${target} LLVM_LTO $<$<CONFIG:Release>:YES>)
set_xcode_property(${target} LLVM_LTO $<$<CONFIG:RelWithDebugInfo>:YES>)
```

Or we could combine the two build modes into a single line using a more complex/ugly [generator expression](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html):
```
set_xcode_property(${target} LLVM_LTO $<$<OR:$<CONFIG:Release>,$<CONFIG:RelWithDebugInfo>>:YES>)
```

This should have the same effect as `-flto` and has the benefit of being reflected in the Build Settings UI:

![screen shot 2018-07-30 at 8 15 17 pm](https://user-images.githubusercontent.com/1198851/43430122-5edaeaf0-9435-11e8-878f-77a624972c69.png)